neo/SmartContract/InteropService.cs

```diff
@@ -200,13 +249,17 @@ private static bool Runtime_CheckWitness(ApplicationEngine engine)
 
         private static bool Runtime_Notify(ApplicationEngine engine)
         {
-            engine.SendNotification(engine.CurrentScriptHash, engine.CurrentContext.EvaluationStack.Pop());
+            StackItem state = engine.CurrentContext.EvaluationStack.Pop();
+            if (!CheckItemForNotification(state)) return false;
```

Comment:
Maybe the best solution is change the VM, because i think that this items comes from the VM, so if we have a problem, should be fixed in the VM.

https://github.com/neo-project/neo-vm/blob/b57c2d304a861692f812a2807f54e8e84534a528/src/neo-vm/ExecutionEngine.cs#L307

to

```csharp
if (!CheckStackSize(false, int.MaxValue) || !OnSysCall(instruction.TokenU32) || !CheckStackSize(false, int.MaxValue))
  return false;
```

Reply:
Notification is not a vm layer thing.

Reply:
NEO3:

Then, I think that we should receive an Array, and in this array, nothing different than ByteArray, Integer or Bool. Is faster and I think that is enough for a notification.